source/base/parameter_handler.cc

```diff
@@ -1174,7 +1219,22 @@ ParameterHandler::print_parameters(std::ostream &    out,
   // we treat XML and JSON is one step via BOOST, whereas all of the others are
   // done recursively in our own code. take care of the two special formats
   // first
-  if (style == XML)
+
+  // explicity eliminate the documentation from the tree if requested
+  if (style == ShortXML || style == ShortJSON)
+    {
+      // create a copy of the tree if it has not been done yet
+      if (sort_alphabetical == false)
+        {
+          sorted_or_simplified_entries = *entries;
+          current_entries              = &sorted_or_simplified_entries;
```

Comment:
This repeated copying and taking the address of here and above is just very difficult to follow.

Just copy the whole damn structure once at the top unconditionally into a new variable. Then you don't need to follow two variables with a pointer pointing to one of the two, and the whole sorting and shrinking code gets easier.

Reply:
I have made the change. 

Regarding your first comment: the function has the same structure as `read_xml_recursively()`; that is why I added the following comment:

```cpp
  // This function is strongly based on read_xml_recursively().
```

I guess the way it is done is safe, since it is done also there.